src/term/mod.rs

```diff
@@ -103,26 +103,33 @@ impl Search for Term {
     }
 
     fn url_search(&self, mut point: Point<usize>) -> Option<String> {
+        let last_col = self.grid.num_cols() - 1;
+
         // Switch first line from top to bottom
         point.line = self.grid.num_lines().0 - point.line - 1;
 
         // Remove viewport scroll offset
         point.line += self.grid.display_offset();
 
         // Create forwards and backwards iterators
-        let iterf = self.grid.iter_from(point);
+        let mut iterf = self.grid.iter_from(point);
         point.col += 1;
         let mut iterb = self.grid.iter_from(point);
 
         // Find URLs
         let mut url_parser = UrlParser::new();
         while let Some(cell) = iterb.prev() {
-            if url_parser.advance_left(cell.c) {
+            if (iterb.cur.col == last_col && !cell.flags.contains(cell::Flags::WRAPLINE))
+                || url_parser.advance_left(cell.c)
+            {
                 break;
             }
         }
-        for cell in iterf {
-            if url_parser.advance_right(cell.c) {
+
+        while let Some(cell) = iterf.next() {
+            if url_parser.advance_right(cell.c)
+                || (iterf.cur.col == last_col && !cell.flags.contains(cell::Flags::WRAPLINE))
```

Comment:
Just a small nit:

Above, you have the `WRAPLINE` check before `advance_left`, and here you have it the other way around. If there's no reason for that I think it would look nicer to keep the same order (so either `WRAPLINE` before `advance_*` in both, or vice versa).

Reply:
This is intentional. The short-circuiting here is used to return *before* the url parser is advanced when moving to the left.